File: converted/utilities/quick-lru/elide-quick-lru.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-quick-lru
#
# This demonstrates calling the TypeScript quick-lru implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One quick-lru implementation shared across TypeScript and Ruby
# - Consistent simple and fast lru cache implementation across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript quick-lru ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# quick_lru_module = Elide.require('./elide-quick-lru.ts')

# Example: Basic usage
# result = quick_lru_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses simple and fast lru cache implementation"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for API response caching, memoization, performance optimization"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide quick-lru (TypeScript)         │"
puts "│   conversions/quick-lru/elide-quick-lru.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same simple and fast lru cache implementation everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/quick_lru.rb"
puts "  QUICK_LRU = Elide.require('./elide-quick-lru.ts')"

File: converted/utilities/quick-lru/elide-quick-lru.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-quick-lru

This demonstrates calling the TypeScript quick-lru implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One quick-lru implementation shared across TypeScript and Python
- Consistent simple and fast lru cache implementation across services
- No Python custom implementation needed
- Perfect for API response caching, memoization, performance optimization
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# quick_lru_module = require('./elide-quick-lru.ts')

print("=== Python Calling TypeScript quick-lru ===\n")

# Example: Basic usage
# result = quick_lru_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses simple and fast lru cache implementation")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for API response caching, memoization, performance optimization")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide quick-lru (TypeScript)         │")
print("│   conversions/quick-lru/elide-quick-lru.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same simple and fast lru cache implementation everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  quick_lru = require('./elide-quick-lru.ts')")
print("  # Use it directly!")
